src/Recipient.c

```diff
@@ -867,6 +874,8 @@ byte * _COSE_RecipientInfo_generateKey(COSE_RecipientInfo * pRecipient, int algI
 	const cn_cbor * pK;
 	byte *pbSecret = NULL;
 
+	UNUSED(algIn);
```

Comment:
algIn is not unused in all cases. I'd like to see a clearer indication of when it is used and when it is not.

Reply:
usages of `algIn` are all included in `#if ... #endif` preprocessor directives. `algIn` will never be used if none of those macros is defined. To make an _unused_ indication, we need to write:
```c
#if !USE_Direct_HKDF_HMAC_SHA_256 && !USE_Direct_HKDF_HMAC_SHA_512 && \
     !USE_Direct_HKDF_AES_128 && !USE_Direct_HKDF_AES_256 && \
     !USE_ECDH_ES_HKDF_256 && !USE_ECDH_ES_HKDF_512 && \
     !USE_ECDH_SS_HKDF_256 && !USE_ECDH_SS_HKDF_512
    UNUSED(algIn);
#endif
```

Do we truly wants this everywhere? 

I think the `UNUSED()` macro should be understood as `MAYBE_UNUSED()`. Or I can rename it like that.

Reply:
I like the idea of a MAYBE_UNUSED(). I think that is more clear. Keep UNUSED() for parameters that are truly never used.